frontend-service/src/components/Game.tsx

```diff
@@ -1,74 +1,79 @@
 import React, { useState, useEffect } from 'react'
 import { fetchRounds } from '../utils/gameUtils'
-import type {Round } from './Game.types'
+import type { CurrentRound, GameState } from './types/Game.types'
+import { gameStatus } from '../objects/gameStatuses'
 import MapboxMap from './MapBoxMap'
 import TopBarGame from './TopBarGame'
 import StartModal from './StartModal'
 import EndModal from './EndModal'
-
+import { indexOfFinalRound } from '../objects/gameConsts'
 
 export default function Game() {
-	const [currentRoundIndex, setCurrentRoundIndex] = useState<number>(0)
-	const [rounds, setRounds] = useState<Round[] | null>(null)
-	const [roundCompleted, setRoundCompleted] = useState<boolean>(false)
-	const [score, setScore] = useState<number>(0)
-	const [isStartModalOpen, setIsStartModalOpen] = useState(true); 
-	const [isEndModalOpen, setIsEndModalOpen] = useState(false);
+	const [currentRound, setCurrentRound] = useState<CurrentRound>({
+		index: 0,
+		completed: false
+	});
+	const [gameState, setGameState] = useState<GameState>({
+		rounds: null,
+		score: 0,
+		status: gameStatus.NOT_STARTED
+	});
```

Comment:
These useStates were pretty dependent on each other and simplifying it down to 2 really makes the entire frontend a lot easier to reason around.